iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -375,32 +378,24 @@ public static void checkRatedU2(Validable validable, double ratedU2) {
         checkRatedU(validable, ratedU2, "2");
     }
 
-    public static ValidationLevel checkSvcRegulator(Validable validable, double voltageSetpoint, double reactivePowerSetpoint,
+    public static ValidationLevel checkSvcRegulator(Validable validable, boolean regulating, double voltageSetpoint, double reactivePowerSetpoint,
                                                     StaticVarCompensator.RegulationMode regulationMode, ValidationLevel validationLevel, ReportNode reportNode) {
-        return checkSvcRegulator(validable, voltageSetpoint, reactivePowerSetpoint, regulationMode, checkValidationLevel(validationLevel), reportNode);
+        return checkSvcRegulator(validable, regulating, voltageSetpoint, reactivePowerSetpoint, regulationMode, checkValidationLevel(validationLevel), reportNode);
     }
 
-    private static ValidationLevel checkSvcRegulator(Validable validable, double voltageSetpoint, double reactivePowerSetpoint,
+    private static ValidationLevel checkSvcRegulator(Validable validable, boolean regulating, double voltageSetpoint, double reactivePowerSetpoint,
                                                     StaticVarCompensator.RegulationMode regulationMode, boolean throwException, ReportNode reportNode) {
-        if (regulationMode == null) {
-            throwExceptionOrLogError(validable, "Regulation mode is invalid", throwException, reportNode);
-            return ValidationLevel.EQUIPMENT;
-        }
-        switch (regulationMode) {
-            case VOLTAGE -> {
-                if (Double.isNaN(voltageSetpoint)) {
-                    throwExceptionOrLogErrorForInvalidValue(validable, voltageSetpoint, VOLTAGE_SETPOINT, throwException, reportNode);
-                    return ValidationLevel.EQUIPMENT;
-                }
-            }
-            case REACTIVE_POWER -> {
-                if (Double.isNaN(reactivePowerSetpoint)) {
-                    throwExceptionOrLogErrorForInvalidValue(validable, reactivePowerSetpoint, "reactive power setpoint", throwException, reportNode);
-                    return ValidationLevel.EQUIPMENT;
-                }
-            }
-            case OFF -> {
-                // nothing to check
+
+        if (regulating) {
+            if (regulationMode == null) {
+                throwExceptionOrLogError(validable, "Svc regulation mode is not set", throwException, reportNode);
+                return ValidationLevel.EQUIPMENT;
```

Comment:
This check should be performed even when `regulating` is false. This will prevent having null values for `regulationMode` in SSH level.